core/src/main/scala/abel/aggregates/Quantile.scala

```diff
@@ -0,0 +1,35 @@
+package abel.aggregates
+import org.isarnproject.sketches.TDigest
+import spray.json.{JsNumber, JsObject, JsValue}
+
+trait Quantile extends Aggregate[Quantile]
+
+case class Quantile1(value: Long) extends Quantile{
+  override def plus(another: Quantile): Quantile = another match {
+    case other:Quantile1 => QuantileN(TDigest.combine(this.toTDigest, other.toTDigest))
+    case QuantileN(otherTDigest) => QuantileN(TDigest.combine(this.toTDigest,otherTDigest))
+  }
+  def toTDigest = {
+    TDigest.sketch(List(this.value))
+  }
+  override def show: JsValue = JsObject("50" -> JsNumber(value),
```

Comment:
Quantile1 represents quantile on a single value whereas QuantileN is on the list of values using TDigest. If we have only a single value for a metric, then quantile would return the same value for all the percentiles. Not sure if this is mathematically right, as the percentile doesn't make any sense on population of size 1. We can return a different message(something like not enough samples) instead of JSNumber or stick to the current so that at least we know what the single value is when we query. Let me know your thoughts.